sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java

```diff
@@ -2728,6 +2764,66 @@ public VirtualMachineImpl withOSDisk(Disk disk) {
         return withOSDisk(disk.id());
     }
 
+    @Override
+    public VirtualMachineImpl withTrustedLaunch() {
+        ensureSecurityProfile().withSecurityType(SecurityTypes.TRUSTED_LAUNCH);
+        return this;
+    }
+
+    @Override
+    public VirtualMachineImpl withSecureBoot() {
+        if (securityType() == null) {
+            return this;
+        }
+        ensureUefiSettings().withSecureBootEnabled(true);
+        return this;
+    }
+
+    @Override
+    public VirtualMachineImpl withoutSecureBoot() {
+        if (securityType() == null) {
+            return this;
+        }
+        ensureUefiSettings().withSecureBootEnabled(false);
+        return this;
+    }
+
+    @Override
+    public VirtualMachineImpl withVTpm() {
+        if (securityType() == null) {
+            return this;
+        }
+        ensureUefiSettings().withVTpmEnabled(true);
+        return this;
+    }
```

Comment:
Does it mean either "TRUSTED_LAUNCH" or "CONFIDENTIAL_VM" can enable security boot and vTPM?

From code seems only "TRUSTED_LAUNCH"?

Reply:
Yes, both can enable secure boot and vTPM. This PR only added support for `TRUSTED_LAUNCH`. 

If we want to support confidential VM, we can add 
```java
WithSecurityFeatures withConfidentialVM();
```
under `WithSecurityTypes` and share secure boot and vTPM settings. Maybe we can add it in the future?

Reply:
Sure, confidential VM can be in future. (I am not sure if it is still a preview feature or stable one)

Reply:
OK, they announced GA this July. Only a few sizes are supported (DCasv5/ECasv5, AMD based which is not in our `VirtualMachineSizeTypes`).
https://azure.microsoft.com/en-us/updates/azureconfidentialvm/

Logged an issue [here](https://github.com/Azure/azure-sdk-for-java/issues/32205) but may implement it when an actual customer requests..